src/plugins/dev_tools/public/application.tsx

```diff
@@ -153,7 +156,12 @@ function DevToolsWrapper({
               mountedTool.current.devTool !== activeDevTool ||
               mountedTool.current.mountpoint !== element)
           ) {
-            await remount(element);
+            let initialDataSourceId;
+            if (!dataSourceEnabled || (dataSourceEnabled && !hideLocalCluster)) {
+              initialDataSourceId = '';
+            }
+
+            await remount(element, initialDataSourceId);
```

Comment:
where do we use the datasourceId passed in here?

Reply:
Thanks for the comment.
The dataSouceId get consumed in `"src/plugins/console/public/application/containers/editor/legacy/console_editor/editor.tsx"`